examples/impls/test/time-sync-avoid-padding.yml

```diff
@@ -0,0 +1,50 @@
+name: nesting-demo
+description: impl with 2 levels of nesting with non-uniform timing of observations, with flag to avoid time-sync by erroring out
+tags:
+initialize:
+  models:
+    - name: time-synchronization
+      model: TimeSyncModel
+      path: builtin
+      config: 
+        start-time: 2023-12-12T00:00:00.000Z # ISO timestamp
+        end-time: 2023-12-12T00:01:00.000Z # ISO timestamp
+        interval: 5 # seconds
+        error-on-padding: true
+graph:
+  children:
+    child: # an advanced grouping node
+      pipeline:
+        - time-synchronization
+      config:
+      children:
+        child-1:
+          inputs:
+            - timestamp: 2023-12-12T00:00:00.000Z
+              duration: 10
+              cpu-util: 10
+              carbon: 100 
+              energy: 100   
+              requests: 300
+            - timestamp: 2023-12-12T00:00:15.000Z
+              duration: 10
+              cpu-util: 20    
+              requests: 300
+            - timestamp: 2023-12-12T00:00:35.000Z
+              duration: 15
+              cpu-util: 20    
+              requests: 400
+        child-2:
+          inputs: 
+            - timestamp: 2023-12-12T00:00:00.000Z
+              duration: 10
+              cpu-util: 10    
+              requests: 300
+            - timestamp: 2023-12-12T00:00:10.000Z
+              duration: 30
+              cpu-util: 20    
+              requests: 380
+            - timestamp: 2023-12-12T00:00:50.000Z
+              duration: 20
+              cpu-util: 20    
+              requests: 380
```

Comment:
newline is missing at the end of file, please check

Reply:
Done